examples/seismic/model.py

```diff
@@ -329,14 +334,16 @@ def __init__(self, origin, spacing, shape, space_order, vp, nbpml=20,
         self.grid = Grid(extent=extent, shape=shape_pml,
                          origin=origin, dtype=dtype)
 
-        # Create square slowness of the wave as symbol `m`
-        if isinstance(vp, np.ndarray):
-            self.m = Function(name="m", grid=self.grid, space_order=space_order)
+        if vp is None:
+            self.m = Function(name="m", grid=self.grid, space_order=self.space_order)
+            try:
+                # Assume m has not yet been padded.
+                initialize_function(self.m, m, self.nbpml)
+            except:
+                # In this case m has already been padded (eg unpickling)
+                self.m.data[:] = m
```

Comment:
Maybe make the try/except part of  `initialize_function` to avoid duplicates. Will also work for TTI parameters then.

Reply:
👍